vmm/src/vm_config.rs

```diff
@@ -115,7 +120,16 @@ pub struct MemoryZoneConfig {
     #[serde(default)]
     pub prefault: bool,
 }
-
+impl MemoryZoneConfig {
+    pub fn apply_landlock(self, landlock: Landlock) -> LandlockResult<Landlock> {
```

Comment:
Why does this function (and other similar ones) need to take a `self`? If you take a reference to then you can skip all the `clone` calls below?

Also, since this is the same function that gets called all the time, you can define a trait as part of the LandLock infrastructure in CH. And then you may do dynamic dispatch without writing code whenever a new configuration option is added.

Reply:
I thought about the second point. It is not really hassle free. Even with a default implementation you still need to tag all the configuration structures. That's a bit tedious.

Reply:
> Why does this function (and other similar ones) need to take a self? If you take a reference to then you can skip all the clone calls below?

This is what I had in my previous version. I pushed this modification in response to comments from Mickael. I agree this is not clean. I will sync up with Mickael on this change before I push my next version.